anchor/network/src/network.rs

```diff
@@ -140,12 +156,24 @@ fn build_anchor_behaviour(local_keypair: Keypair) -> AnchorBehaviour {
         .unwrap();
 
     let gossipsub =
-        gossipsub::Behaviour::new(MessageAuthenticity::Signed(local_keypair), config).unwrap();
+        gossipsub::Behaviour::new(MessageAuthenticity::Signed(local_keypair.clone()), config)
+            .unwrap();
+
+    let discovery = {
+        // Build and start the discovery sub-behaviour
+        let mut discovery = Discovery::new(local_keypair.clone(), network_config)
+            .await
+            .unwrap();
```

Comment:
nit, either:
```suggestion
            .expect("Discovery should be able to start");
```
or `?` to bubble up the error and treat it in the function caller

Reply:
we'll need to refactor it and return a `Result` instead.

Reply:
Could we create an issue and address it in a new PR?

Reply:
yeah ofc :)

Reply:
https://github.com/sigp/anchor/issues/75